src/java/com/aap/gitst/Logger_copy.java

```diff
@@ -0,0 +1,340 @@
+
+                    printProgress();
+                } else {
+                    _logWriter.println(msg);
+
+                    if (ex != null) {
+                        ex.printStackTrace(_logWriter);
+                    }
+                }
+
+                _logWriter.flush();
+            }
+        }
+    }
+
+    public ProgressBar createProgressBar(final Object message, final int total) {
+        return createProgressBar(message, total, false);
+    }
+
+    public ProgressBar createProgressBar(final Object message, final int total,
+            final boolean closeOnComplete) {
+        if (!isProgressBarSupported() || (total == 0)) {
+            return new DummyProgressBar();
+        }
+
+        synchronized (this) {
+            final PBar pbar = new PBar(message, total, closeOnComplete);
+            _pbar.add(pbar);
+            pbar.start();
+            return pbar;
+        }
+    }
+
+    public boolean isDebugEnabled() {
+        return getLevel() == Level.DEBUG;
+    }
+
+    public boolean isInfoEnabled() {
+        return getLevel().ordinal() <= Level.INFO.ordinal();
+    }
+
+    public boolean isWarnEnabled() {
+        return getLevel().ordinal() <= Level.WARNING.ordinal();
+    }
+
+    public boolean isErrorEnabled() {
+        return getLevel().ordinal() <= Level.ERROR.ordinal();
+    }
+
+    private synchronized void log(final Object msg) {
+        if (!_pbar.isEmpty()) {
+            clearPogress();
+            _logWriter.println(msg);
+            printProgress();
+        } else {
+            _logWriter.println(msg);
+        }
+
+        _logWriter.flush();
+    }
+
+    // @GuardedBy("this")
+    private void clearPogress() {
+        _logWriter.print('\r');
+        for (final PBar b : _pbar) {
+            for (int i = 0; i < b._lastMessageLen; i++) {
+                _logWriter.print(' ');
+            }
+            _logWriter.print(' ');
+        }
+        _logWriter.print('\r');
+    }
+
+    // @GuardedBy("this")
+    private void printProgress() {
+        for (final PBar b : _pbar) {
+            b.printMessage();
+            _logWriter.print(' ');
+        }
+        _logWriter.flush();
+    }
+
+    public static enum Level {
+        DEBUG,
+        INFO,
+        WARNING,
+        ERROR,
+        OFF;
+    }
+
+    public static interface ProgressBar {
+
+        public void done(int count);
+
+        public void update();
+
+        public void complete();
+
+        public void close();
+    }
+
+    public static class DummyProgressBar implements ProgressBar {
+
+        @Override
+        public void done(final int count) {
+        }
+
+        @Override
+        public void update() {
+        }
+
+        @Override
+        public void complete() {
+        }
+
+        @Override
+        public void close() {
+        }
+    }
+
+    public class PBar extends Thread implements ProgressBar {
+        private final Object _message;
+        private final int _total;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this field "_total" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS00116)